bids_conversion/convert_somato_data.py

```diff
@@ -0,0 +1,222 @@
+"""Convert the MNE-somato-data to BIDS format.
+
+Requirements
+------------
+The python dependencies to run this script are:
+- numpy
+- scipy
+- mne<0.19
+- mne-bids>0.2
+- nibabel
+
+Notes on freesurfer
+-------------------
+We *cannot* simply copy over the freesurfer files from the non-BIDS somato
+data, because we need to rename the subject to '01' instead of 'somato'. See
+the example workflow below:
+
+$ export SUBJECTS_DIR=/home/username/subjects
+
+$ echo $SUBJECTS_DIR
+/home/username/subjects
+
+$ recon-all -all -i ~/MNE-somato-data-bids/sub-01/anat/sub-01_T1w.nii.gz -s 01
+
+Then your `$SUBJECTS_DIR` should contain a folder `01` with the somato MRI data
+derivatives and you can successfully run the following cell.
+
+Helpful links
+-------------
+- https://mne-tools.github.io/stable/index.html
+- https://mne-tools.github.io/mne-bids/
+- https://mne-tools.github.io/stable/manual/datasets_index.html?#somatosensory
+- https://surfer.nmr.mgh.harvard.edu/fswiki/FreeSurferWiki
+
+"""
+# Authors: Stefan Appelhoff <stefan.appelhoff@mailbox.org>
+#
+# License: BSD (3-clause)
+import os
+import os.path as op
+import shutil as sh
+import json
+import warnings
+
+import numpy as np
+import mne
+import mne.datasets.somato as somato
+from mne_bids import write_raw_bids, make_bids_basename, write_anat
+from mne_bids.utils import print_dir_tree
+
+# Somato data prior to conversion
+somato_path = somato.data_path()
+
+# Print the directory tree
+print('\nSomato data before conversion:\n')
+print_dir_tree(somato_path, maxdepth=3)
+print('\n\n')
+
+# Convert to BIDS
+# Path to the raw data of somato dataset
+fif_path = op.join(somato_path, 'MEG', 'somato')
+
+# Load the raw data file
+raw = mne.io.read_raw_fif(op.join(fif_path, 'sef_raw_sss.fif'))
```

Comment:
This script will actually not work anymore once you update the dataset. Any way to make it more reproducible?

Reply:
I think it WILL work, because I am **requiring** MNE 0.18 ... and that version of MNE will point to the version of the MNE-somato-data that is not BIDS

Reply:
oh I see, as long as the old data is not removed from OSF :-) sounds good. Maybe you can have a one line version check above ...

Reply:
sounds good, thanks!